actions/src/project_actions.py

```diff
@@ -34,20 +34,39 @@ def run(self, submodule: str, **kwargs):
         return func(**kwargs)
 
     def project_delete(
-        self, cloud_account: str, project_identifier: str
+        self, cloud_account: str, project_identifier: str, delete: bool
     ) -> Tuple[bool, str]:
         """
         Deletes a project
         :param cloud_account: The account from the clouds configuration to use
         :param project_identifier: (Either) The project name to delete
+        :param delete: When true will delete the project, when false will only return it to reduce
+                       chances of accidental deletion
         :return: The result of the operation
         """
-        delete_ok = self._identity_api.delete_project(
+        if delete:
+            delete_ok = self._identity_api.delete_project(
```

Comment:
So this is logic creeping into our actions layer

What would be better is if `delete_project` took the flag, then run all steps except for the real call. This also is trivial to unit test so we can have a dedicated test for it.

That way it truly is a dry run rather than two logic paths dependent on a flag.